java-util/src/main/java/io/druid/java/util/http/client/pool/ChannelResourceFactory.java

```diff
@@ -109,6 +112,28 @@ public ChannelFuture generate(final String hostname)
 
       final ChannelPipeline pipeline = connectFuture.getChannel().getPipeline();
       pipeline.addFirst("ssl", sslHandler);
+      pipeline.addLast("connectionErrorHandler", new SimpleChannelUpstreamHandler()
```

Comment:
`addLast` adds this handler at the last of the pipeline, but it should be right after `sslHandler` which is added at the first of the pipeline.

Reply:
Hi, @jihoonson . What you mean is that we should use the following code to ensure that the order of those handlers is `SSL handler + Error handler + Others handlers`?

```java
pipeline.addFirst("connectionErrorHandler", new SimpleChannelUpstreamHandler(){...})
pipeline.addFirst("ssl", sslHandler);
```

IMO, maybe the current order is fine, the `Error handler` should be placed at the end of the pipeline, so that the order of the handlers will be `SSL handler + Others handler + Error handlers`. What do you think?

Reply:
Sounds good. Thanks.